systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java

```diff
@@ -218,38 +230,40 @@ void testUserTemplate(ExtensionContext extensionContext) {
         assertThat(userSecret.getMetadata().getAnnotations().get(annotationKey), is(annotationValue));
     }
 
-    synchronized void testUserWithQuotas(ExtensionContext extensionContext, KafkaUser user) {
-        String userName = KafkaUserResource.kafkaUserClient().inNamespace(NAMESPACE).withName(user.getMetadata().getName()).get().getStatus().getUsername();
+    synchronized void testUserWithQuotas(ExtensionContext extensionContext, KafkaUser user, String clusterName) {
+        final String namespaceName = StUtils.getNamespaceBasedOnRbac(NAMESPACE, extensionContext);
 
-        Integer prodRate = 1111;
-        Integer consRate = 2222;
-        Integer reqPerc = 42;
-        Double mutRate = 10d;
+        final Integer prodRate = 1111;
+        final Integer consRate = 2222;
+        final Integer reqPerc = 42;
+        final Double mutRate = 10d;
 
         // Create user with correct name
         resourceManager.createResource(extensionContext, KafkaUserTemplates.userWithQuotas(user, prodRate, consRate, reqPerc, mutRate)
             .editMetadata()
-                .withNamespace(NAMESPACE)
+                .withNamespace(namespaceName)
             .endMetadata()
             .build());
 
+        final String userName = KafkaUserResource.kafkaUserClient().inNamespace(namespaceName).withName(user.getMetadata().getName()).get().getStatus().getUsername();
+
         String command = "bin/kafka-configs.sh --bootstrap-server localhost:9092 --describe --entity-type users";
         LOGGER.debug("Command for kafka-configs.sh {}", command);
 
-        ExecResult result = cmdKubeClient(NAMESPACE).execInPod(KafkaResources.kafkaPodName(userClusterName, 0), "/bin/bash", "-c", command);
+        ExecResult result = cmdKubeClient(namespaceName).execInPod(KafkaResources.kafkaPodName(clusterName, 0), "/bin/bash", "-c", command);
         assertThat(result.out().contains("Quota configs for user-principal '" + userName + "' are"), is(true));
         assertThat(result.out().contains("request_percentage=" + reqPerc), is(true));
         assertThat(result.out().contains("producer_byte_rate=" + prodRate), is(true));
         assertThat(result.out().contains("consumer_byte_rate=" + consRate), is(true));
         assertThat(result.out().contains("controller_mutation_rate=" + mutRate), is(true));
```

Comment:
Not sure it has to be fixed in this PR and maybe it is not an issue at all and I just missed somehting. But what I wonder about is:
* You run all the tests in parallel against the same cluster, so there will be multiple users
* Your `command` lists quotas for all users as it seems

So, how do you know that for example `assertThat(result.out().contains("request_percentage=" + reqPerc), is(true))` actually asserts quota from this particular user and not from another user?

Reply:
That's a good point. I think I should create a separate Kafka cluster for each KafkaUser. (I can open another PR to fix that...)

Reply:
Can't we somehow add the username to the command instead? From the perspective of the execution time it is of course great to share the clusters for this ... we just need to make sure the quotas are evaluated for given user.

Reply:
https://github.com/strimzi/strimzi-kafka-operator/pull/5584